drivers/util.py

```diff
@@ -1098,24 +1097,30 @@ def dom0_disks():
     return disks
 
 
-def set_scheduler_sysfs_node(node, str):
-    """Set the scheduler for a sysfs node (e.g. '/sys/block/sda')"""
+def set_scheduler_sysfs_node(node, scheds):
+    """
+    Set the scheduler for a sysfs node (e.g. '/sys/block/sda')
+    according to prioritized list schedulers
+    Try to set the first item, then fall back to the next on failure
+    """
 
     path = os.path.join(node, "queue", "scheduler")
     if not os.path.exists(path):
         SMlog("no path %s" % path)
         return
-    try:
-        f = open(path, 'w')
-        f.write("%s\n" % str)
-        f.close()
-        SMlog("Set scheduler to [%s] on [%s]" % (str, node))
-    except:
-        SMlog("Error setting scheduler to [%s] on [%s]" % (str, node))
-        pass
+    for sched in scheds:
+        try:
+            with open(path, 'w') as file:
+                file.write("%s\n" % sched)
+            SMlog("Set scheduler to [%s] on [%s]" % (sched, node))
+            return
+        #pylint: disable=broad-except
+        except Exception as err:
+            SMlog("Warning setting scheduler to [%s] on [%s] with [%s]" % (sched, node, str(err)))
```

Comment:
If there are multiple schedulers to try this probably shouldn't be a warning, otherwise as discussed since schedulers are pluggable it'd be difficult to detect what is available at runtime other than trying and failing.

Perhaps 'none' vs 'noop' should be detectable, but not the others.